src/scanning/ScanPage.tsx

```diff
@@ -0,0 +1,69 @@
+import React, { useState, useEffect } from 'react';
+import { Text, Container, Heading, Box, Spinner } from 'theme-ui';
+import QrReader from 'react-qr-reader';
+import { useHistory } from 'react-router-dom';
+import { createAccessPass } from '../api';
+import { useAuthentication } from '../authentication';
+
+const uuidValidationRegex = /^[0-9a-f]{8}-[0-9a-f]{4}-[1-5][0-9a-f]{3}-[89ab][0-9a-f]{3}-[0-9a-f]{12}$/i;
+
+const errorClearTimeout = 5000;
+
+export const ScanPage = () => {
+  const history = useHistory();
+  const { userId, token } = useAuthentication();
+  const [error, setError] = useState<string | null>(null);
+  const [loading, setLoading] = useState(false);
+
+  async function handleScan(data: string | null) {
+    if (data && userId && token && !loading) {
+      // TODO: embed a piece of info before uuid (cov-user?) to make sure it's our uuid
```

Comment:
I think this is a good idea, but I would make it domain specific.  If we use QRs for tracking posted tests, we will probably be verifying them against a different endpoint.  It would be good to understand which type of QR we're scanning so we can go and do the right thing with it.

Reply:
Exactly, thought the same